aip/aog/3021.md

```diff
@@ -0,0 +1,64 @@
+---
+aip:
+  id: 3021
+  state: draft
+  created: 2019-10-04
+  updated: 2019-10-04
+  type: guidance
+  scope: aog
+js:
+  - /assets/js/graphviz/viz.js
+  - /assets/js/graphviz/lite.render.js
+  - /assets/js/aip/aip-graphviz.js
+permalink: /3021
+---
+
+# Type protos in Actions on Google APIs
+
+Many Actions on Google fulfillment APIs need to represent typed data (datetime,
+money, etc.). In some cases, there are common components for representing these
+types.
+
+## Guidance
+
+Fulfillment APIs **must** use the AoG common types when applicable, and **must
+not** use alternative type protos. For example, an API which needs to represent
+a monetary amount **must** use [google.type.Money][money], and **must not** use
+an alternative representation.
+
+APIs **may** wrap existing types in new protos. This may be useful in order to
+define compound types, provide additional metadata, provide semantic signals
+(if the typed value was obtained from a user query), etc.
+
+## Actions on Google common types
+
+The AoG common types include all the protobuf types and API types defined in
+[AIP-213][], as well as all the protos in
+[google.actions.type.\*][actions-type].
+
+### Adding new types
+
+If a given API needs a type for which there is no existing common type, and the
+type is not conceptually specific to that API, and the type is well understood
+enough to be modeled in a permanent/unversioned way, then API author **should**
+consider proposing a new AoG common type in
+[google.actions.type.\*][actions-type].
+
+The guidance in the appendix of [AIP-213][] also applies to creating new AoG
+common types. Specifically, API authors proposing new types **should** consider
+whether a new type should be a Google-wide common type rather than an AoG
+common type. One exception to this is that new AoG common types are not
+proposed by opening a GitHub ticket. Instead, AoG API authors should see
+//google/actions/type/OWNERS for contact information.
```

Comment:
Not a fan of having internal paths in a public AIP. Can this be done with a GitHub ticket, or some other means?

Reply:
It's a fair sentiment.  But the existing process is internal, and moving it
to GitHub just for the sake of one line of an AIP doesn't seem worth it.
How about rewording this instead?  And/or including an internal mailing
list?

Reply:
Seems fair.

Reply:
On reflection, including an internal-only mailing list didn't seem right either, so I reworded this to remove the reference to a specific file.  WDYT?